pkg/api/v1/helper/helpers.go

```diff
@@ -29,11 +29,11 @@ import (
 	"k8s.io/kubernetes/pkg/api/helper"
 )
 
-// IsExtendedResourceName returns true if the resource name is not in the
+// IsGenericResourceName returns true if the resource name is not in the
```

Comment:
What is a generic resource? Instead have an integer resource list for first class resources and use that.

Reply:
+1, integer resource predicate can match all extended resources (for now) and the integral first-class resources including hugepages.